homeassistant/core.py

```diff
@@ -1281,11 +1256,9 @@ def __init__(
         self.data: _DataT = data or {}  # type: ignore[assignment]
         self.origin = origin
         self.time_fired_timestamp = time_fired_timestamp or time.time()
-        if not context:
-            context = Context(id=ulid_at_time(self.time_fired_timestamp))
-        self.context = context
-        if not context.origin_event:
-            context.origin_event = self
```

Comment:
Context may be created in other places.  This looks like a semantic change to not set here